lib/reader/src/error.rs

```diff
@@ -40,7 +41,7 @@ macro_rules! err {
     ( $loc:expr, $msg:expr ) => {
         Err($crate::Error {
             location: $loc.clone(),
-            message: String::from($msg),
+            message: ::std::string::String::from($msg),
```

Comment:
The issue is that it's in a macro. Maybe it was something else that I was running into, but as far as I could tell, it needed to have `::std::string::String` because when compiling without `std`, String isn't automagically included into each file.